### DISCUSSION TITLE
DRAMClocks with Intel Optane DC Persistent Memory

### DISCUSSION THREAD

**Discussion by AnaKhorguani** [opcm/pcm#359] [Q&A]
Hello, I am using information from ./pcm-power.x for power usage evaluation. 

I am experimenting on servers equipped with Intel Optane DC persistent memory (PMEM), configured in App Direct mode, which puts this persistent memory alongside DRAM, as part of the main memory. 

With such environment, does DRAMClocks still report about the frequency of DRAM, or is there a possibility it could be reporting the memory frequency of PMEM instead? 

Thank you in advance.

**Comment by opcm** [comment]
I am not aware of a performance monitoring event for reporting memory frequency of PMEM.

**Comment by twillhal** [comment]
Ana, thanks for bringing this to our attention. What range of frequency are you observing? Please note that, depending on your configuration, DRAM frequency might drop when Intel Optane PMem is installed. For example, a system might run at 3200 MT/s with only DRAM installed, but only at 2933 MT/s with DRAM and Optane PMem installed.

**Reply by AnaKhorguani29** [reply]
Hello, thank you for your reply. It is an interesting remark but I have a bit different case. 

I have two machines: A - with only DRAM, and B - with DRAM and Optane PMem. For both, configured Memory Speed is 2666 MT/s. Given its Double Data Rate characteristic, I  expected measured frequency to be 1333 MT/s.

When memory is accessed, `DRAMClocks` reports 1333 MT/s for both machines. However, when memory is idle, `DRAMClocks` for A is just several tens of MHz (e.g. 46.8 MHz), while for machine B, `DRAMClocks` stays 1333 MT/s.

When machines are idle, `CKE Off Residency` reports for both, that DRAM is in power down mode. Given that some internal clocks might be disabled in such modes, I thought this could explain 46.8 MHz frequency. However, then it is inconsistent with `DRAMClocks` reporting 1333 MT/s for idle DRAM in machine B, which is also in power down mode. Given that in machine B there is also Optane PMem, that is where I started wondering whether `DRAMClocks` reported DRAM or PMem frequency.  

Of course, there is a possibility that DRAM behavior is simply different in machines that are also equipped with Optane PMem, but still I thought it could be helpful to double check that `DRAMClocks` indeed reports DRAM frequency. 

I apologize for my prolonged discussion, and I hope it does not leave the scope of the topic.
